pkg/splunk/enterprise/configuration.go

```diff
@@ -308,11 +306,41 @@ func validateCommonSplunkSpec(spec *enterpriseApi.CommonSplunkSpec) error {
 		return fmt.Errorf("Negative value (%d) is not allowed for Readiness probe intial delay", spec.ReadinessInitialDelaySeconds)
 	}
 
+	// if not provided, set default values for imagePullSecrets
+	err := ValidateImagePullSecrets(ctx, c, cr, spec)
+	if err != nil {
+		return err
+	}
+
 	setVolumeDefaults(spec)
 
 	return ValidateSpec(&spec.Spec, defaultResources)
 }
 
+// ValidateImagePullSecrets sets default values for imagePullSecrets if not provided
+func ValidateImagePullSecrets(ctx context.Context, c splcommon.ControllerClient, cr splcommon.MetaObject, spec *enterpriseApi.CommonSplunkSpec) error {
+	reqLogger := log.FromContext(ctx)
+	scopedLog := reqLogger.WithName("ValidateImagePullSecrets").WithValues("name", cr.GetName(), "namespace", cr.GetNamespace())
+
+	// If no imagePullSecrets are configured
+	var nilImagePullSecrets []corev1.LocalObjectReference
+	if len(spec.ImagePullSecrets) == 0 {
+		spec.ImagePullSecrets = nilImagePullSecrets
+		return nil
+	}
+
+	// If configured, validated if the secret/s exist
+	for _, secret := range spec.ImagePullSecrets {
+		_, err := splutil.GetSecretByName(ctx, c, cr.GetNamespace(), cr.GetName(), secret.Name)
+		if err != nil {
+			scopedLog.Error(err, "Secret in the imagePullSecrets config is not found ", secret.Name)
```

Comment:
Just thinking aloud.. if the secret in imagePullSecrets is not found, should we just war, use default values and continue?

Reply:
Good point @smohan-splunk , the K8s client runs through the list of secrets to find a valid secret. The only problem would be when we don't find any of the secrets! Let me change the logic to throw an error only if NONE of them are found.